### DISCUSSION TITLE
404 Error with ChatGPT 4 (but not with 3.5)

### DISCUSSION THREAD

**Discussion by AlessioBona** [chhoumann/quickadd#564] [Q&A]
I am getting errors when trying to use ChatGPT 4, but the same command, same prompt, they work, as long as I use ChatGPT 3.5. Both ChatGPT 4 ad 4 32k fail.

I have changed my API key to a new one, but nothing changed.

My account on OpenAI works and chaptGPT 4 works there.

The error says:
caught (in promise) Error: Error while making request to OpenAI API: Request failed, status 404
    at makeRequest (plugin:quickadd:11190:13)

I really hope someone can help, because I adore using ChatGPT directly from inside Obsidian <3

**Comment by pustovit** [comment]
Same here, but doesn't work for GPT-3 either. GPT-4 returns a 404, GPT-3 returns a 429. Would love to see a workaround

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

Both errors here were on OpenAI's side, not QuickAdd's. In 2023, GPT-4 API access was gated: accounts without approved/paid API usage got a 404 (model not available to that key) even when GPT-4 worked fine in the ChatGPT web app, since ChatGPT and the API were separate products. The 429 @pustovit saw is the quota/no-credit variant of the same thing.

The AI Assistant has been rewritten since then: you configure your own providers and model lists per provider, so there's no baked-in 2023 model set anymore. Docs: https://quickadd.obsidian.guide/docs/AIAssistant

Closing as outdated. If anyone hits provider errors on a current version, please open a fresh issue with the provider and model you're using.
